httpserver/handlers.go

```diff
@@ -174,9 +176,9 @@ func (server *Server) mutate(ctx context.Context, w http.ResponseWriter, r *http
 			}()
 			parsedReference, err := pkgUtils.ParseSubjectReference(image)
 			if err != nil {
-				errMessage := fmt.Sprintf("failed to mutate image reference %s: %v", image, err)
-				logrus.Error(errMessage)
-				returnItem.Error = errMessage
+				err = errors.ErrorCodeReferenceInvalid.WithError(err).WithDetail(fmt.Sprintf("failed to parse image reference %s", image))
```

Comment:
Wondering if its easier if we had a wrapper method errors.getRatifyError( ErrorCodeReferenceInvalid, componentType, pluginName, err, detailedMessage)

Reply:
sure, I created a constructor which can be used if we need to assign more than 2 fields.